bolt-sidecar/src/state/consensus.rs

```diff
@@ -40,6 +84,37 @@ impl ConsensusState {
 
         self.header = update.header.message;
 
+        // Update the timestamp with current time
+        self.timestamp = current_timestamp();
+
+        // Get the current value of slot and epoch
+        let slot = self.header.slot;
+        let epoch = slot / 32;
+
+        // If the epoch has changed, update the proposer duties
+        if epoch != self.epoch.value {
+            self.epoch.value = epoch;
+            self.epoch.start_slot = epoch * 32;
+
+            self.fetch_proposer_duties(epoch).await?;
+        }
+
         Ok(())
     }
+
+    /// Fetch proposer duties for the given epoch.
+    async fn fetch_proposer_duties(&mut self, epoch: u64) -> Result<(), ConsensusError> {
+        let duties = self.beacon_api_client.get_proposer_duties(epoch).await?;
+
+        self.epoch.proposer_duties = duties.1;
+        Ok(())
+    }
+}
+
+/// Get the current timestamp.
+fn current_timestamp() -> u64 {
+    return SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
```

Comment:
Just a util function, ideally should be in utils.rs (not sure of the file structure in rust)

Reply:
NOTE - the precision of the deadline is in seconds. We might want to do it in mili seconds or smaller